webserver/webserver/src/main/java/io/helidon/webserver/UpgradeCodecSupplier.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.helidon.webserver;
+
+import java.util.Optional;
+
+/**
+ * Service providing HTTP upgrade codec for Helidon webserver.
+ */
+public interface UpgradeCodecSupplier {
```

Comment:
Is this a service provider (as in loaded by service loader?). If yes, it must be in `spi` package.